### PR TITLE
Enable more nullable reference types in asn.xslt.

### LINKED ISSUE #33824: Some .xml.cs do not match nullability when re-generated
I noticed that in two cases, running `msbuild /t:CompileAsn` against S.S.C.Pkcs/src in master results in differences despite the XML file having not changed.

Specifically:

```diff
diff --git a/src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Asn1/Rfc3161TimeStampReq.xml.cs b/src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Asn1/Rfc3161TimeStampReq.xml.cs
index d31c8fc0236..530d1f36ca7 100644
--- a/src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Asn1/Rfc3161TimeStampReq.xml.cs
+++ b/src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Asn1/Rfc3161TimeStampReq.xml.cs
@@ -18,7 +18,7 @@ internal partial struct Rfc3161TimeStampReq
 
         internal int Version;
         internal System.Security.Cryptography.Pkcs.Asn1.MessageImprint MessageImprint;
-        internal Oid? ReqPolicy;
+        internal Oid ReqPolicy;
         internal ReadOnlyMemory<byte>? Nonce;
         internal bool CertReq;
         internal System.Security.Cryptography.Asn1.X509ExtensionAsn[] Extensions;
diff --git a/src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Asn1/SignerInfoAsn.xml.cs b/src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Asn1/SignerInfoAsn.xml.cs
index dc2267293d8..e381bbfa7f6 100644
--- a/src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Asn1/SignerInfoAsn.xml.cs
+++ b/src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Asn1/SignerInfoAsn.xml.cs
@@ -20,7 +20,7 @@ internal partial struct SignerInfoAsn
         internal ReadOnlyMemory<byte>? SignedAttributes;
         internal System.Security.Cryptography.Asn1.AlgorithmIdentifierAsn SignatureAlgorithm;
         internal ReadOnlyMemory<byte> SignatureValue;
-        internal System.Security.Cryptography.Asn1.AttributeAsn[]? UnsignedAttributes;
+        internal System.Security.Cryptography.Asn1.AttributeAsn[] UnsignedAttributes;
 
         internal void Encode(AsnWriter writer)
         {
```

Steps to reproduce:

1. Check out master (f80a5147ef7662c3238275568c9dad7655a9684f)
2. cd `./src/libraries/System.Security.Cryptography.Pkcs/src`
3. `msbuild /t:CompileAsn`
4. `git diff` shows the diff above.

/cc @bartonjs (sorry I am noisy today)

### PR DESCRIPTION
**Pull request by vcsjones**
* string and Oid types on Choice can be null.
* Reference types that are `@optional` can be null.

Fixes #33824 

/cc @bartonjs

### CODE REVIEW

File: src/libraries/Common/src/System/Security/Cryptography/Asn1/AlgorithmIdentifierAsn.xml.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
```
Comment (vcsjones) [inline src/libraries/Common/src/System/Security/Cryptography/Asn1/AlgorithmIdentifierAsn.xml.cs]: Most of these `#nullable enable` comes from the `asn.xslt`, this is generated code.

File: src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Rfc3161TimestampRequest.cs [hunk 2]
```diff
@@ -38,7 +38,7 @@ public X509ExtensionCollection GetExtensions()
                 return coll;
             }
 
-            X509ExtensionAsn[] rawExtensions = _parsedData.Extensions;
+            X509ExtensionAsn[] rawExtensions = _parsedData.Extensions!;
```
Comment (vcsjones) [inline src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Rfc3161TimestampRequest.cs]: The `if (!HasExtensions)` on line 36 checks that `Extensions` is not null and > 0.

File: src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/UnixPkcs12Reader.cs [hunk 3]
```diff
@@ -248,7 +248,7 @@ private void Decrypt(ReadOnlySpan<char> password, ReadOnlyMemory<byte> authSafeC
             // Nothing requires that there be fewer keys than certs,
             // but it's sort of nonsensical when loading this way.
             CertBagAsn[] certBags = ArrayPool<CertBagAsn>.Shared.Rent(10);
-            AttributeAsn[][] certBagAttrs = ArrayPool<AttributeAsn[]>.Shared.Rent(10);
+            AttributeAsn[]?[] certBagAttrs = ArrayPool<AttributeAsn[]?>.Shared.Rent(10);
```
Comment (vcsjones) [inline src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/UnixPkcs12Reader.cs]: Line 688 could potentially assign null to an index in `certBagAttrs`

File: src/libraries/Common/src/System/Security/Cryptography/Asn1/asn.xslt [hunk 4]
```diff
@@ -51,6 +51,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
```
Comment (stephentoub) [inline src/libraries/Common/src/System/Security/Cryptography/Asn1/asn.xslt]: Why is this necessary? Are these files included in a project that's not yet annotated?

Comment (vcsjones) [reply src/libraries/Common/src/System/Security/Cryptography/Asn1/asn.xslt]: It may not be; one of the other XSLT templates in the same file had `#nullable enable`, so some of the files already had them:

https://github.com/dotnet/runtime/blob/419e949d258ecee4c40a460fb09c66d974229623/src/libraries/Common/src/System/Security/Cryptography/Asn1/asn.xslt#L130

Would it be preferable to get rid of all of them if possible now?

Comment (stephentoub) [reply src/libraries/Common/src/System/Security/Cryptography/Asn1/asn.xslt]: If they're no longer necessary, yes, please.

Comment (vcsjones) [reply src/libraries/Common/src/System/Security/Cryptography/Asn1/asn.xslt]: There are 5 projects that have `AsnXml` in them and they are have `<nullable>enable</nullable>`, so it appears that `#nullable enable` does not need to be in the template anymore.

Comment (stephentoub) [reply src/libraries/Common/src/System/Security/Cryptography/Asn1/asn.xslt]: Cool.

**Review by stephentoub — APPROVED** [review]
Thanks

File: src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/UnixExportProvider.cs [hunk 5]
```diff
@@ -197,25 +197,26 @@ private byte[] ExportPfx(SafePasswordHandle password)
                 attrBytes[1] = sizeof(int);
                 MemoryMarshal.Write(attrBytes.AsSpan(2), ref keyIdx);
 
+                AttributeAsn attribute = new AttributeAsn
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/UnixExportProvider.cs]: Why did this need to move out?

File: src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/UnixExportProvider.cs [hunk 6]
```diff
@@ -197,25 +197,26 @@ private byte[] ExportPfx(SafePasswordHandle password)
...
-                        new AttributeAsn
-                        {
-                            AttrType = new Oid(Oids.LocalKeyId, null),
-                            AttrValues = new ReadOnlyMemory<byte>[]
-                            {
-                                attrBytes,
-                            }
-                        }
+                        attribute
                     }
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/UnixExportProvider.cs]: Either

```C#
BagAttributes = new[] { attribute },
```

or


```C#
BagAttributes = new[]
{
    attribute,
},
```

, please.  (That's "Add the trailing comma, or collapse it to one line")

File: src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/UnixExportProvider.cs (hunk 5 above)
Comment (bartonjs) [reply src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/UnixExportProvider.cs]: Ah, presumably to avoid nullable annotations in the indirect reference it used to have.

**Review by bartonjs — APPROVED** [review]
